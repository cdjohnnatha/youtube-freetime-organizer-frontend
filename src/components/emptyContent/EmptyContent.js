import React from 'react';
import { string } from 'prop-types';
import { Typography, Avatar, Grid } from '@material-ui/core';
import SentimentDissatisfiedIcon from '@material-ui/icons/SentimentDissatisfied';

const styles = {
  mainGridContainerStyle: {
    padding: '3em',
  },
  avatarContainerStyle: {
    marginBottom: '2em',
  }
}

const EmptyContent = ({ message }) => {
  return (
    <Grid
      container
      direction="column"
      justify="center"
      alignItems="center"
      style={styles.mainGridContainerStyle}
    >
      <Grid style={styles.avatarContainerStyle}>
        <Avatar>
          <SentimentDissatisfiedIcon />
        </Avatar>
      </Grid>
      <Grid>
        <Typography>{message}</Typography>
      </Grid>
    </Grid>
  )
}

EmptyContent.propTypes = {
  message: string,
}

EmptyContent.defaultProps = {
  message: 'It was not found any content for this resource',
}

export default EmptyContent;
