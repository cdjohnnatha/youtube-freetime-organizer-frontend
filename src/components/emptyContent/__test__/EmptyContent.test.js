
import React from 'react';
import { create } from 'react-test-renderer';
import EmptyContent from '../EmptyContent';


describe('Test EmptyContent component', () => {
  test('EmptyContent snapshot renders', () => {
    const component = create(<EmptyContent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('EmptyContent component', () => {
    const component = create(
      <EmptyContent />
    ).root;
    const emptyContent = component.findByType('div');
    expect(component.props.message).toBe('It was not found any content for this resource');
    expect(emptyContent.type).toBe('div');
  });
});