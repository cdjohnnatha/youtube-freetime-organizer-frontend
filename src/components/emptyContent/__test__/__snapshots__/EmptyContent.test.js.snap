// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EmptyContent component EmptyContent snapshot renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  style={
    Object {
      "padding": "3em",
    }
  }
>
  <div
    className="MuiGrid-root"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="15.5"
          cy="9.5"
          r="1.5"
        />
        <circle
          cx="8.5"
          cy="9.5"
          r="1.5"
        />
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-3.5c.73 0 1.39.19 1.97.53.12-.14.86-.98 1.01-1.14-.85-.56-1.87-.89-2.98-.89-1.11 0-2.13.33-2.99.88.97 1.09.01.02 1.01 1.14.59-.33 1.25-.52 1.98-.52z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      It was not found any content for this resource
    </p>
  </div>
</div>
`;
